### PR TITLE
Integrate CodeRx MDT maintenance_inhaler submodule into asthma module

### LINKED ISSUE #1181: Increase maintenance inhaler medication diversity in asthma module
### Requested Feature

The asthma module currently only prescribes a single maintenance inhaler product to 100% of the population, regardless of age.  This is not consistent with real-world clinical practice.  Also (note) the RXCUI prescribed does not actually relate to any active NDCs. 
 Would like to create a more representative maintenance inhaler medication profile - suggestion to use CodeRx Medication Diversification Tool (MDT).

### PR DESCRIPTION
**Pull request by jrlegrand**
Resolves #1181 

Created a `maintenance_inhaler` submodule following the CodeRx Medication Diversification Tool (MDT) tip sheet at https://coderx.io/projects/medication-diversification#tip-sheet.

Integrated the submodule into Synthea.

Ran a population of 5000 patients in Synthea and examined the CSV output to verify the diversity of maintenance inhaler medications having REASONDESCRIPTION "Asthma" / "Childhood Asthma" has increased from one to many.

Attaching the log file output ([log 20221024-062521.txt](https://github.com/synthetichealth/synthea/files/9851851/log.20221024-062521.txt)) from the [CodeRx Medication Diversification Tool (MDT)](https://github.com/coderxio/medication-diversification) - it helps summarize the expected distributions of  medication products at a population level and also provides a snapshot of the settings used to create this MDT submodule.

This is the first submission using the [CodeRx Medication Diversification Tool (MDT)](https://github.com/coderxio/medication-diversification) but we plan on submitting several more in the near future!

### CODE REVIEW

**Comment by jawalonoski** [comment]
Thank you for the pull request! Clinical review was good, so we'll do some technical checks and then merge this in soon.

**Comment by jawalonoski** [comment]
Looks like some of the unit tests broke:

```
Caused by:
[2254](https://github.com/synthetichealth/synthea/actions/runs/3312999002/jobs/5475000835#step:6:2255)
        java.lang.IllegalArgumentException: resource modules/lookup_tables/maintenance_inhaler_Beclomethasone_product_distribution.csv not found.
```

Looking into the cause...

**Comment by jrlegrand** [comment]
I think the B in Beclomethasone got capitalized when it should be all lower case. Prob something that could be fixed in MDT but also could be fixed here to make the PR work. I'm traveling now but just wanted to point this out.

**Comment by jawalonoski** [comment]
> I think the B in Beclomethasone got capitalized when it should be all lower case. Prob something that could be fixed in MDT but also could be fixed here to make the PR work. I'm traveling now but just wanted to point this out.

The case does appear to be the issue, but I can't recreate the test failure on either Mac OS X or Windows... and data generation on both platforms also works as expected... yet it fails consistently on the build.

**Comment by dehall** [comment]
I think it's a difference in file system case sensitivity. Some file systems will consider "file.txt" and "File.txt" to be the same but others consider them different. I seem to remember seeing this with some MDT modules I've experimented with as well - the lookup table files are named with capitals in some places but are referenced in the modules with lower case.

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+                "-- --------- ----",
+                "1. [ 44.7% ] Breath_Actuated_120_Actuat_Beclomethasone_Dipropionate_0_04_Mg_Actuat_Metered_Dose_Inhaler_Qvar",
+                "2. [ 55.3% ] Breath_Actuated_120_Actuat_Beclomethasone_Dipropionate_0_08_Mg_Actuat_Metered_Dose_Inhaler_Qvar"
+            ],
+            "type": "Simple",
+            "lookup_table_transition": [
+                {
+                    "transition": "Prescribe_Breath_Actuated_120_Actuat_Beclomethasone_Dipropionate_0_04_Mg_Actuat_Metered_Dose_Inhaler_Qvar",
+                    "default_probability": "1",
+                    "lookup_table_name": "maintenance_inhaler_Beclomethasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:155]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_beclomethasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_Breath_Actuated_120_Actuat_Beclomethasone_Dipropionate_0_08_Mg_Actuat_Metered_Dose_Inhaler_Qvar",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Beclomethasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:160]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_beclomethasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+                "4. [ 10.1% ] Budesonide_0_125_Mg_Ml_Inhalation_Suspension_Pulmicort",
+                "5. [ 49.5% ] Budesonide_0_25_Mg_Ml_Inhalation_Suspension",
+                "6. [ 2.4% ] Budesonide_0_5_Mg_Ml_Inhalation_Suspension"
+            ],
+            "type": "Simple",
+            "lookup_table_transition": [
+                {
+                    "transition": "Prescribe_120_Actuat_Budesonide_0_18_Mg_Actuat_Dry_Powder_Inhaler_Pulmicort",
+                    "default_probability": "1",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:185]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_60_Actuat_Budesonide_0_09_Mg_Actuat_Dry_Powder_Inhaler_Pulmicort",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:190]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_Budesonide_0_125_Mg_Ml_Inhalation_Suspension",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:195]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_Budesonide_0_125_Mg_Ml_Inhalation_Suspension_Pulmicort",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:200]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 7]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_Budesonide_0_25_Mg_Ml_Inhalation_Suspension",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:205]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 8]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_Budesonide_0_5_Mg_Ml_Inhalation_Suspension",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Budesonide_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:210]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_budesonide_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 9]
```diff
@@ -0,0 +1,725 @@
...
+                "6. [ 3.4% ] 60_Actuat_Fluticasone_Propionate_0_05_Mg_Actuat_Dry_Powder_Inhaler_Flovent",
+                "7. [ 5.5% ] 60_Actuat_Fluticasone_Propionate_0_1_Mg_Actuat_Dry_Powder_Inhaler_Flovent",
+                "8. [ 0.5% ] 60_Actuat_Fluticasone_Propionate_0_25_Mg_Actuat_Dry_Powder_Inhaler_Flovent"
+            ],
+            "type": "Simple",
+            "lookup_table_transition": [
+                {
+                    "transition": "Prescribe_120_Actuat_Fluticasone_Propionate_0_044_Mg_Actuat_Metered_Dose_Inhaler_Flovent",
+                    "default_probability": "1",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:237]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 10]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_120_Actuat_Fluticasone_Propionate_0_11_Mg_Actuat_Metered_Dose_Inhaler_Flovent",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:242]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 11]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_120_Actuat_Fluticasone_Propionate_0_22_Mg_Actuat_Metered_Dose_Inhaler_Flovent",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:247]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 12]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_30_Actuat_Fluticasone_Furoate_0_1_Mg_Actuat_Dry_Powder_Inhaler_Arnuity",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:252]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 13]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_30_Actuat_Fluticasone_Furoate_0_2_Mg_Actuat_Dry_Powder_Inhaler_Arnuity",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:257]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 14]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_60_Actuat_Fluticasone_Propionate_0_05_Mg_Actuat_Dry_Powder_Inhaler_Flovent",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:262]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 15]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_60_Actuat_Fluticasone_Propionate_0_1_Mg_Actuat_Dry_Powder_Inhaler_Flovent",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:267]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 16]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_60_Actuat_Fluticasone_Propionate_0_25_Mg_Actuat_Dry_Powder_Inhaler_Flovent",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Fluticasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:272]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_fluticasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 17]
```diff
@@ -0,0 +1,725 @@
...
+                "3. [ 27.9% ] 120_Actuat_Mometasone_Furoate_0_2_Mg_Actuat_Metered_Dose_Inhaler_Asmanex",
+                "4. [ 3.1% ] 14_Actuat_Mometasone_Furoate_0_22_Mg_Actuat_Dry_Powder_Inhaler_Asmanex",
+                "5. [ 33.8% ] 60_Actuat_Mometasone_Furoate_0_22_Mg_Actuat_Dry_Powder_Inhaler_Asmanex"
+            ],
+            "type": "Simple",
+            "lookup_table_transition": [
+                {
+                    "transition": "Prescribe_120_Actuat_Mometasone_Furoate_0_1_Mg_Actuat_Metered_Dose_Inhaler_Asmanex",
+                    "default_probability": "1",
+                    "lookup_table_name": "maintenance_inhaler_Mometasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:296]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_mometasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 18]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_120_Actuat_Mometasone_Furoate_0_22_Mg_Actuat_Dry_Powder_Inhaler_Asmanex",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Mometasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:301]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_mometasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 19]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_120_Actuat_Mometasone_Furoate_0_2_Mg_Actuat_Metered_Dose_Inhaler_Asmanex",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Mometasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:306]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_mometasone_product_distribution.csv"
```

File: src/main/resources/modules/medications/maintenance_inhaler.json [hunk 20]
```diff
@@ -0,0 +1,725 @@
...
+                },
+                {
+                    "transition": "Prescribe_14_Actuat_Mometasone_Furoate_0_22_Mg_Actuat_Dry_Powder_Inhaler_Asmanex",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Mometasone_product_distribution.csv"
+                },
+                {
+                    "transition": "Prescribe_60_Actuat_Mometasone_Furoate_0_22_Mg_Actuat_Dry_Powder_Inhaler_Asmanex",
+                    "default_probability": "0",
+                    "lookup_table_name": "maintenance_inhaler_Mometasone_product_distribution.csv"
```
Comment (jawalonoski) [inline src/main/resources/modules/medications/maintenance_inhaler.json:316]:
```suggestion
                    "lookup_table_name": "maintenance_inhaler_mometasone_product_distribution.csv"
```

**Review by jawalonoski — CHANGES_REQUESTED** [review]
It looks like the build environment file names are case-sensitive. I recommend that you keep the all-lowercase filenames and change the JSON lookups to be all-lowercase as well.

**Pull request merged**